### DISCUSSION TITLE
Deploy the hugo project into github pages with github actions. but encounter error "failed to load modules"

### DISCUSSION THREAD

**Discussion by xhddx** [gethinode/hinode#784] [Help] [answered]
**Describe the bug**
I try to deploy the hugo project into github pages with github actions. but encounter error "failed to load modules" 

**To reproduce**
1. add  .github/workflows/hugo.yaml to loacl repo and push to remote
2. Click on gitHub  'Action' to check the process
3. I see failed result

**Expected behavior**
hugo \
    --gc \
    --minify \
    --baseURL "https://teholic.github.io/"
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    HUGO_VERSION: 0.122.0
    GITHUB_PAGES: true
    HUGO_ENVIRONMENT: production
    HUGO_ENV: production
hugo: downloading modules …
hugo: collected modules in 18[6](https://github.com/teholic/teholic.github.io/actions/runs/7969793655/job/21756202599#step:7:6)66 ms
Total in 1866[7](https://github.com/teholic/teholic.github.io/actions/runs/7969793655/job/21756202599#step:7:7) ms
Error: failed to load modules: module "hinode" not found in "/home/runner/work/teholic.github.io/teholic.github.io/themes/hinode"; either add it as a Hugo Module or store it in "/home/runner/work/teholic.github.io/teholic.github.io/themes".: module does not exist
Error: Process completed with exit code 1.

**Log file**

**Screenshots**

**Environment (please complete the following information):**
 - Hinode version: v 0.22.5
 - Host OS: ubuntu-latest
 - Node version: 20.11.1
 - Hugo version: hugo v0.122.0-b9a03bd59d5f71a529acb3e33f995e0ef332b3aa+extended darwin/arm64 BuildDate=2024-01-26T15:54:24Z VendorInfo=brew
 - Browser: [e.g. Google Chrome Version 108.0.5359.124 (Official Build) (arm64)]

**Additional context**
[Add any other context about the problem here.]

**Comment by markdumay** [comment]
It's difficult to reproduce the error with the information given, but I suspect you need to vendor your modules prior to the build. Try running `hugo mod vendor` prior to `hugo --gc --minify --baseURL "https://teholic.github.io/"`.

If it still fails, would you mind sharing a (minimal) repo or the exact steps to reproduce the error? In particular, your custom action `.github/workflows/hugo.yaml` would be helpful, as well as if you are using Hinode as a regular theme or with npm.

**Reply by xhddx** [reply]
yes, how to share a minimal repo?

**Reply by xhddx** [reply]
the yaml file is copyed from here
https://gohugo.io/hosting-and-deployment/hosting-on-github/#procedure
and I just modify the hugo version from 0.123.0 to 0.122.0. nothing else changed.

**Reply by xhddx** [reply, marked as answer]
I finally figure it out,  before I push my hugo paoject to remote, I use command hugo to compile my project, which generate a "public" folder, that cause this problem. so no need to perform hugo before push to git remote.

**Reply by markdumay** [reply]
ok, glad you solved it!
